src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/ReflectionJsonTypeInfoOfT.cs

```diff
@@ -20,6 +20,11 @@ internal ReflectionJsonTypeInfo(JsonConverter converter, JsonSerializerOptions o
             : base(converter, options)
         {
             NumberHandling = GetNumberHandlingForType(Type);
+            if (Kind == JsonTypeInfoKind.Object)
+            {
+                UnmappedMemberHandling = GetUnmappedMemberHandling(Type);
```

Comment:
It seems the below two method calls are also only applicable to object type-kinds.

Reply:
`PopulatePolymorphismMetadata` also works for collection types. You're right about `MapInterfaceTypesToCallbacks`, although in the future we should extend callback support to collection types as well.

Reply:
See https://github.com/dotnet/runtime/issues/71945